apps/astarte_pairing/lib/astarte_pairing/queries.ex

```diff
@@ -117,6 +117,34 @@ defmodule Astarte.Pairing.Queries do
     end
   end
 
+  def verify_already_registered_device(realm_name, device_id) do
```

Comment:
In my opinion it would be more correct, since it returns `{:ok, boolean}`, to change the function to `verify_already_registered_device?(...)`. 
Also, the function is a bit too specific... might it be worth changing the pattern/logic to simply use `select_device_by_id(realm_name, device_id)` and manage externally whether it is present/registered in the db or not?

Reply:
On function name: we usually follow the `check_*` convention for `{:ok, bool}` results (see e.g. `check_device_exists/2` in RealmManagement), good catch!


On function logic: code that is already in place uses the pattern of selecting only relevant information (see e.g. `select_device_for_credentials_request/2` and `select_device_for_info/2`), I have followed the same pattern. Your suggestion would be interesting when refactoring Pairing!
Also, note that a function that performs the same query does already exist (`check_already_registered_device/2`), however it uses CQEx and therefore it needs a CQEx client rather than a simple realm name  to run. I decided to keep `engine.ex` code simple and duplicate the functionality in `queries.ex` - not only for that, but also because we will soon™ remove CQEx entirely.